### ISSUE TITLE
Ignore specific packages

### ISSUE THREAD

**Issue by benoitmasson** [kardianos/govendor#156]
Hi, I'd like to know if there is a way to explicitly exclude some packages from the `govendor` scope. I don't think that the `ignore` field of `vendor.json` currently allows this (or I couldn't find the way to do it)… Wouldn't it be interesting to add this possibility?

I am aware that with usual dependencies, it is rather useless to exclude a particular `github.com/…` dependency from being vendored, but my use case is different.

I have the following code structure in my `$GOPATH`:
```
src ┬ app ┬ main.go   (imports lib)
    │     └ vendor ┈
    │
    ├ lib ┬ lib.go    (imports github.com/a/b)
    ┊     └ vendor ┈
    ┊
    └ github.com ─ a ─ b ┈
```

So, `govendor list` in the `app` directory reports that `lib`, and thus by transitivity `github.com/a/b`, are external dependencies. This is true, but as my lib evolves a lot along with my app, I don't want to vendor it and its dependencies in my app.

If I could tell `govendor` to ignore the `lib` folder, then `govendor list` wouldn't list any external dependency anymore, as expected in this case.

**Comment by kardianos** [comment]
I understand why you might want to do this.

However, I don't think it would work. If your app/cmd and lib reference the same package, then they would need packages to have the same path. As you develop your lib, my advice is to either: clone directly into the app/cmd sub-folder or just vendor after you've released a new lib feature set.

lib/vendor/github/a/a
app/vendor/github/a/a

Both "a" packages, but compiler will treat them as totally different. If both app and lib depend on the same package, you will very likely have trouble down the road.

**Comment by benoitmasson** [comment]
> If your app/cmd and lib reference the same package, then they would need packages to have the same path

You're right, however, I believe this is the case only when both pieces of code reference variables or unexported definitions in the same imported package. Functions and type definitions should work without trouble.

I actually faced this issue earlier, it was caused by the use of `reflect` in one project, to get an interface object implemented by an unexported type initialized in the other project (thus, rather unconventional)... which I solved by moving the first package to the other side, in the same place as its dependencies. I think that it is more a conception issue, that I got wrong at first, that would cause this `govendor` side-effect.

> My advice is to either: clone directly into the app/cmd sub-folder or just vendor after you've released a new lib feature set.

Probably... but I'd rather not vendor too often my app's dependencies, and the lib evolves a lot. As for me, the best way around this is a “global” `vendor` folder rooted at `src`, that both `app` and `lib` would use. But `govendor` is not compatible with that scheme (`src` is not a project directory), and it would re-couple `app` and `lib` dependencies...

Anyway, apart from my particular problem, this may not be the only use-case for the initial request... Don't you think adding an `exclude` field to `vendor.json`, with the list of packages to exclude from vendoring, would be useful?

These could be listed with a new status `+excluded (x)` for example, so that the `govendor list` user can still see them and distinguish them from regular external packages...
